src/components/TodoForm/TodoForm.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useState } from 'react';
+import { Todo } from '../../types/Todo';
+import usersFromServer from '../../api/users';
+import { getUserById } from '../../services/user';
+
+type Props = {
+  onSubmit: (todo: Todo) => void;
+};
+
+export const TodoForm: React.FC<Props> = ({ onSubmit }) => {
+  const [title, setTitle] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+
+  const [userId, setUserId] = useState(0);
+  const [hasUserIdError, setHasUserIdError] = useState(false);
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setHasTitleError(false);
+  };
+
+  const handleUserIdChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setHasUserIdError(false);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    setHasTitleError(!title);
+    setHasUserIdError(!userId);
+
+    if (!title || !userId) {
+      return;
+    }
+
+    onSubmit({
+      id: 0,
+      title,
+      completed: false,
+      userId,
+      user: getUserById(userId),
```

Comment:
Using `getUserById(userId)` here is appropriate for assigning a user to the todo. However, ensure that this function does not return `null` or `undefined`, as this could lead to issues when rendering the todo list. This aligns with checklist item 4, which advises against rendering components with `null` or `undefined` values.